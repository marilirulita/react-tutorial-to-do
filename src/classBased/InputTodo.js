import React, { Component } from "react"

class InputTodo extends Component {
  state = {
    title: ""
  }

  handleOnChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    }) 
  }

  handleSubmit = e => {
    e.preventDefault();
    if (this.state.title.trim()) {
      this.props.handleSubmitProps(this.state.title);
    this.setState({
      title: ""
    })
    }
    else {
      alert('Please write an item')
    }
    
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit} className="form-container">
        <input type="text" className="input-text" placeholder="Add Todo..." name="title" value={this.state.title} onChange={this.handleOnChange} />
        <button className="input-submit">Submit</button>
      </form>
    )
  }
}
export default InputTodo