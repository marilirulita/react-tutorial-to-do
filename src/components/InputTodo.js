import React, { Component } from "react"

class InputTodo extends Component {
  state = {
    title: ""
  }

  handleOnChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    }) 
  }

  handleSubmit = e => {
    e.preventDefault();
    if (this.state.title.trim()) {
      this.props.handleSubmitProps(this.state.title);
    this.setState({
      title: ""
    })
    }
    else {
      alert('Please write an item')
    }
    
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input type="text" placeholder="Add Todo..." name="title" value={this.state.title} onChange={this.handleOnChange} />
        <button>Submit</button>
      </form>
    )
  }
}
export default InputTodo