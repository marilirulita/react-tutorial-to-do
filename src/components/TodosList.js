import React from "react";
import TodoItem from "./TodoItem";

class TodosList extends React.Component {

  render() {
    return (
      <ul>
        {this.props.todos.map(elem => (
          <TodoItem key={elem.id} todo={elem} handleChangeProps={this.props.handleChangeProps} handleDeleteProps={this.props.handleDeleteProps} handleEditProp={this.props.handleEditProp} />
        ))}
      </ul>
    )      
  }
    
}

export default TodosList;