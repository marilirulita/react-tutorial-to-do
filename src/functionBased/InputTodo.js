import React, { useState } from "react"

const InputTodo = (props) => {
  const [title, setTitle] = useState("");
  const handleOnChange = e => {
    setTitle(e.target.value) 
  }

  const handleSubmit = e => {
    e.preventDefault();
    if (title.trim()) {
      props.handleSubmit(title);
    setTitle("")
    }
    else {
      alert('Please write an item')
    } 
  }

    return (
      <form onSubmit={handleSubmit} className="form-container">
        <input type="text" className="input-text" placeholder="Add Todo..." name="title" value={title} onChange={handleOnChange} />
        <button className="input-submit">Submit</button>
      </form>
    )
}
export default InputTodo