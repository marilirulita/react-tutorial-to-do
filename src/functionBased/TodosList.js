import React from "react";
import TodoItem from "./TodoItem";

const TodosList = (props) => {
  return (
    <ul>
      {props.todos.map((elem) => (
        <TodoItem
          key={elem.id}
          todo={elem}
          handleChange={props.handleChange}
          handleDelete={props.handleDelete}
          handleEdit={props.handleEdit}
        />
      ))}
    </ul>
  );
};

export default TodosList;
