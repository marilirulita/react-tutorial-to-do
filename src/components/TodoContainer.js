import React from "react";
import { v4 as uuidv4 } from "uuid";
import Header from "./Header";
import TodosList from "./TodosList";
import InputTodo from "./InputTodo";

class TodoContainer extends React.Component {
  state = {
    todos: [
      {
        id: uuidv4(),
        title: "Setup development environment",
        completed: false
      },
      {
        id: uuidv4(),
        title: "Develop website and add content",
        completed: false
      },
      {
        id: uuidv4(),
        title: "Deploy to live server",
        completed: false
      }
    ]
   };

   handleChange = id => {
    this.setState(prevState => ({
      todos: prevState.todos.map(todo => {
        if (todo.id === id){
          return {
            ...todo,
            completed: !todo.completed,
          }
        }
        return todo;
      }),
    }))
   }

   handleDelete = id => {
     this.setState({
       todos: [
         ...this.state.todos.filter(todo => todo.id !== id)
       ] 
     })
   }

   addNewItem = title => {
     const newItem = {
      id: uuidv4(),
      title: title,
      completed: false
     }
     this.setState({
       todos: [...this.state.todos, newItem]
     })
   }

  render() {
    return (
      <div className="container">
        <div className="inner">
          <Header />
          <InputTodo handleSubmitProps={this.addNewItem} />
          <TodosList todos={this.state.todos} handleChangeProps={this.handleChange} handleDeleteProps={this.handleDelete} />
        </div>
      </div>
    )
  }
}

export default TodoContainer