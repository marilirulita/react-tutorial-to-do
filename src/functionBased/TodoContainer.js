import React, { useState, useEffect } from "react";
//import { Route, Switch } from "react-router-dom";
import { Routes, Route } from "react-router-dom";

import { v4 as uuidv4 } from "uuid";
import Header from "./Header";
import TodosList from "./TodosList";
import InputTodo from "./InputTodo";
import About from "./pages/About";
import NotMatch from "./pages/NotMatch";
import Navbar from "./Navbar";

const TodoContainer = () => {
  const getInitialTodos = () => {
    const temp = localStorage.getItem("todos");
    const savedTodos = JSON.parse(temp);
    return savedTodos || [];
  };
  const [todos, addTodos] = useState(getInitialTodos());

  const handleChange = (id) => {
    addTodos((prevState) =>
      prevState.map((todo) => {
        if (todo.id === id) {
          return {
            ...todo,
            completed: !todo.completed,
          };
        }
        return todo;
      })
    );
  };

  const handleDelete = (id) => {
    addTodos([...todos.filter((todo) => todo.id !== id)]);
  };

  const addNewItem = (title) => {
    const newItem = {
      id: uuidv4(),
      title: title,
      completed: false,
    };
    addTodos([...todos, newItem]);
  };

  const handleEdit = (updateTitle, id) => {
    addTodos(
      todos.map((todo) => {
        if (todo.id === id) {
          todo.title = updateTitle;
        }
        return todo;
      })
    );
  };

  useEffect(() => {
    const temp = JSON.stringify(todos);
    localStorage.setItem("todos", temp);
  }, [todos]);

  return (
    <>
      <Navbar />
      {/* <Switch> */}
      <Routes>
        <Route exact path="/">
          <div className="container">
            <div className="inner">
              <Header />
              <InputTodo handleSubmit={addNewItem} />
              <TodosList
                todos={todos}
                handleChange={handleChange}
                handleDelete={handleDelete}
                handleEdit={handleEdit}
              />
            </div>
          </div>
        </Route>
        <Route path="/about">
          <About />
        </Route>
        <Route path="*">
          <NotMatch />
         </Route>
         </Routes>
       {/* </Switch> */}
    </>
  );
};

export default TodoContainer;
