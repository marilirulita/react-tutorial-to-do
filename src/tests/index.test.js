import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../functionBased/Header';
import InputTodo from '../functionBased/InputTodo';
import TodoContainer from '../functionBased/TodoContainer';
import TodosList from '../functionBased/TodosList'
//import TodoItem from '../functionBased/TodoItem';

it('renders correctly', () => {
  const tree = renderer
    .create(<Header></Header>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<TodoContainer></TodoContainer>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  
  const tree = renderer
    .create(<InputTodo></InputTodo>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  const tree = renderer
    .create(<TodosList todos={todos}
      handleChange={handleChange}
      handleDelete={handleDelete}
      handleEdit={handleEdit}></TodosList>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});