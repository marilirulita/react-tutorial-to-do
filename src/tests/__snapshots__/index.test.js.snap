// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  style={
    Object {
      "lineHeight": "1.5em",
      "padding": "20px 0",
    }
  }
>
  <h1
    style={
      Object {
        "color": "#ececec",
        "fontSize": "6rem",
        "fontWeight": "600",
        "lineHeight": "1em",
        "marginBottom": "2rem",
        "textAlign": "center",
        "textTransform": "lowercase",
      }
    }
  >
    todos
  </h1>
</header>
`;

exports[`renders correctly 2`] = `
<header
  style={
    Object {
      "lineHeight": "1.5em",
      "padding": "20px 0",
    }
  }
>
  <h1
    style={
      Object {
        "color": "#ececec",
        "fontSize": "6rem",
        "fontWeight": "600",
        "lineHeight": "1em",
        "marginBottom": "2rem",
        "textAlign": "center",
        "textTransform": "lowercase",
      }
    }
  >
    todos
  </h1>
</header>
`;

exports[`renders correctly 3`] = `
<form
  className="form-container"
  onSubmit={[Function]}
>
  <input
    className="input-text"
    name="title"
    onChange={[Function]}
    placeholder="Add Todo..."
    type="text"
    value=""
  />
  <button
    className="input-submit"
  >
    Submit
  </button>
</form>
`;
